allennlp/modules/scalar_mix.py

```diff
@@ -0,0 +1,59 @@
+from typing import List
+
+import torch
+from torch.nn import ParameterList, Parameter
+
+from allennlp.common.checks import ConfigurationError
+
+class ScalarMix(torch.nn.Module):
+    """
+    Computes a parameterised scalar mixture of N tensors, ``mixture = gamma * sum(s_k * tensor_k)``
+    where ``s = softmax(w)``, with ``w`` and ``gamma`` scalar parameters.
+
+    In addition, if ``do_layer_norm=True`` then apply layer normalization to each tensor
+    before weighting.
+    """
+    def __init__(self, mixture_size: int, do_layer_norm: bool = False) -> None:
+        super(ScalarMix, self).__init__()
+
+        self.mixture_size = mixture_size
+        self.do_layer_norm = do_layer_norm
+
+        self.scalar_parameters = ParameterList([Parameter(torch.FloatTensor([0.0]))
+                                                for _ in range(mixture_size)])
+        self.gamma = Parameter(torch.FloatTensor([1.0]))
+
+    def forward(self, tensors: List[torch.Tensor],  # pylint: disable=arguments-differ
```

Comment:
Please can you add a tiny docstring here and explain that the mask is for all of the tensors that are being mixed.